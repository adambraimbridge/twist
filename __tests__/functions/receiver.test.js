/* global expect,jest */
/* eslint-disable no-console */

// This block must go first
global.console = {log: jest.fn()};
const lambdaTester = require('lambda-tester');
const receiver = require('../../functions/receiver');

// You can edit after here.
const payloadFromCircleCI = require('../../__mocks__/fixtures/payload-from-circleci.json');

describe('functions/receiver', () => {
	describe('given an event notification', () => {
		test('it should trigger a Web Page Test', () => {
			return lambdaTester(receiver.handler)
				.event(payloadFromCircleCI)
				.expectResult(response => {
					expect(response).toMatchSnapshot();
				});
		});
		test('it should log to console', () => {
			return lambdaTester(receiver.handler)
				.event(payloadFromS3)
				.expectResult(() => {
					expect(console.log).toBeCalledWith('');
				});
		});
	});
});
