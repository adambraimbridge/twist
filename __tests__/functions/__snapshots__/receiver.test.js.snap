// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`functions/receiver`] = `
Object {
  "body": "Twist: Sucessfully processed payload",
  "statusCode": 200,
}
`;
